src/test/java/org/folio/permstest/RestVerticleTest.java

```diff
@@ -696,34 +980,21 @@ public void testGroup(TestContext context) throws InterruptedException, Executio
 
   private void send(String url, TestContext context, HttpMethod method, String content,
                     String contentType, Handler<HttpClientResponse> handler) {
```

Comment:
Duplicated code should be extracted into methods for get, post, put, delete. A few examples:
```
private Response send(String tenant, HttpMethod method, String path, String content, TestContext context) {
  try {
    CompletableFuture<Response> futureResponse = new CompletableFuture();
    send(tenant, "http://localhost:" + port + path, context, method, content,
          SUPPORTED_CONTENT_TYPE_JSON_DEF, new HTTPResponseHandler(futureResponse));
    return futureResponse.get(5, TimeUnit.SECONDS);
  } catch (Exception e) {
    context.fail(e);
    return null;
  }
}

private Response get(String path, TestContext context) {
  return send("diku", HttpMethod.GET, path, null, context);
}

private Response get(String tenant, String path, TestContext context) {
  return send(tenant, HttpMethod.GET, path, null, context);
}

private Response put(String path, JsonObject content, TestContext context) {
  return send("diku", HttpMethod.PUT, path, content.encode(), context);
}

private Response put(String path, String content, TestContext context) {
  return send("diku", HttpMethod.PUT, path, content, context);
}

private Response put(String tenant, String path, JsonObject content, TestContext context) {
  return send(tenant, HttpMethod.PUT, path, content.encode(), context);
}

```

Reply:
Yes. It's quite bad.. Something like restassured would be better. Can we postpone this to MODINVSTOR-83?

Reply:
Yes, postpone to MODPERMS-83. I don't think that restassured would be better.